src/serde/de.rs

```diff
@@ -395,22 +387,14 @@ impl<'a, 'de> de::Deserializer<'de> for ValueDeserializer<'a, 'de> {
 
     fn deserialize_enum<V>(
         self,
-        name: &'static str,
+        _name: &'static str,
         _variants: &'static [&'static str],
         visitor: V,
     ) -> Result<V::Value, Self::Error>
     where
         V: Visitor<'de>,
     {
-        use IonType::*;
-        if self.value.annotations().next() != Some(Ok(name.as_symbol_ref())) {
-            return IonResult::decoding_error("expected an instance of enum {name}");
-        }
-        match self.value.ion_type() {
-            Symbol => visitor.visit_enum(UnitVariantAccess::new(self)),
```

Comment:
(Question) Shouldn't we still be using this `Symbol` match for `UnitVariantAccess`? I see that the tests are passing, so just want to understand how this change works.

Reply:
This check now happens in `deserialize_identifier`, which helps the enum visitor determine which variant it's working with:
https://github.com/amazon-ion/ion-rust/blob/2f97e99914f4a05100008e6d2945b1456239c5c4/src/serde/de.rs#L408

It now only requires it to be a symbol when it is unannotated, as it's possible for an enum to be defined like this:
```rust
enum Foo {
    Unit,                    // Serializes as an unannotated symbol
    NewType(ion_rs::Symbol), // Serializes as an annotated symbol
}
```
and we don't want the dispatch to see an `IonType::Symbol` enum variant and assume it's a unit type.

Reply:
Nice. Thanks for explaining! Looks good to me.